applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp

```diff
@@ -0,0 +1,1292 @@
+    num_nodes = GetGeometry().size();
+    local_size = num_nodes * 3;
+
+    if (rElementalDofList.size() != local_size)
+        rElementalDofList.resize(local_size);
+
+    SizeType local_index = 0;
+
+    for (SizeType i_node = 0; i_node < num_nodes; ++i_node)
+    {
+        rElementalDofList[local_index++] = this->GetGeometry()[i_node].pGetDof(DISPLACEMENT_X);
+        rElementalDofList[local_index++] = this->GetGeometry()[i_node].pGetDof(DISPLACEMENT_Y);
+        rElementalDofList[local_index++] = this->GetGeometry()[i_node].pGetDof(DISPLACEMENT_Z);
+    }
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement::Initialize()
+{
+    KRATOS_TRY;
+
+    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(GetIntegrationMethod());
+
+    //Constitutive Law initialisation
+    if ( mConstitutiveLawVector.size() != integration_points.size() )
+        mConstitutiveLawVector.resize( integration_points.size() );
+
+    if ( GetProperties()[CONSTITUTIVE_LAW] != nullptr ) {
+        const GeometryType& r_geometry = GetGeometry();
+        const Properties& r_properties = GetProperties();
+        const auto& N_values = r_geometry.ShapeFunctionsValues(GetIntegrationMethod());
+        for ( IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number ) {
+            mConstitutiveLawVector[point_number] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
+            mConstitutiveLawVector[point_number]->InitializeMaterial( r_properties, r_geometry, row(N_values , point_number ));
+        }
+    } else
+        KRATOS_ERROR << "A constitutive law needs to be specified for the element with ID " << this->Id() << std::endl;
+
+    KRATOS_CATCH( "" )
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement::CalculateLeftHandSide(
+    MatrixType& rLeftHandSideMatrix,
+    ProcessInfo& rCurrentProcessInfo)
+
+{
+    TotalStiffnessMatrix(rLeftHandSideMatrix,GetGeometry().GetDefaultIntegrationMethod());
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement::CalculateRightHandSide(
+    VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo)
+
+{
+    const SizeType number_of_nodes = GetGeometry().size();
+    const SizeType dimension = GetGeometry().WorkingSpaceDimension();
+    const SizeType system_size = number_of_nodes * dimension;
+
+    Vector internal_forces = ZeroVector(system_size);
+    InternalForces(internal_forces,GetGeometry().GetDefaultIntegrationMethod());
+    rRightHandSideVector = ZeroVector(system_size);
```

Comment:
you miss the noaliases. please take a good look through the element

Reply:
I hope I got them all now